src/storage/store/property_statistics.cpp

```diff
@@ -0,0 +1,45 @@
+#include "storage/store/property_statistics.h"
+
+#include "common/ser_deser.h"
+#include "storage/store/table_statistics.h"
+
+namespace kuzu {
+namespace storage {
+
+void PropertyStatistics::serialize(common::FileInfo* fileInfo, uint64_t& offset) {
+    common::SerDeser::serializeValue(mayHaveNullValue, fileInfo, offset);
+}
+
+RWPropertyStats::RWPropertyStats(TablesStatistics* tablesStatistics, common::table_id_t tableID,
+    common::property_id_t propertyID)
+    : tablesStatistics{tablesStatistics}, tableID{tableID}, propertyID{propertyID} {}
+
+std::unique_ptr<PropertyStatistics> PropertyStatistics::deserialize(
+    common::FileInfo* fileInfo, uint64_t& offset) {
+    bool hasNull;
+    common::SerDeser::deserializeValue<bool>(hasNull, fileInfo, offset);
+    return std::make_unique<PropertyStatistics>(hasNull);
+}
+
+// In a read-only context, readStatistics should always be non-null
+// Read/write statistics cannot be cached since functions like checkpointInMemoryIfNecessary may
+// overwrite them and invalidate the reference
+bool RWPropertyStats::mayHaveNull() {
```

Comment:
I don't think we should make assumptions on whether to read from the read only version or the write version in this function. Instead, we should take `Transaction*` as a param, and switch based on the transaction type.

Consider this: Merge `TablesStatistics::getWritePropertyStatisticsForTable` and `TablesStatistics::getReadPropertyStatisticsForTable` into a single function `TablesStatistics::getPropertyStatisticsForTable` and that function will take in a transaction pointer as a param. Internally, we will return property statistics based on the transaction type. Other interfaces in `TablesStatistics` should also be refactored to follow this fashion later.

Reply:
I think that's a good idea. The `Transaction` wasn't actually being passed to the `NodeColumn` when I started working on this, and I wasn't really clear to me how to determine when to use one or the other.

Reply:
Using a dummy write transaction gets it compiling, but it doesn't seem to work properly and the tests are failing.

Reply:
The concurrency fix commit has fixed this now